src/Matterhorn/State/Messages.hs

```diff
@@ -850,6 +873,31 @@ notifyGetPayload NotifyV3 st post mentioned = do
                 PostTypeChannelDeleted    -> "channelDeleted"
                 PostTypeEphemeral         -> "ephemeral"
                 PostTypeUnknown _         -> "unknown"
+notifyGetPayload NotifyV4 st post mentioned = do
+    let notification = NotificationV4 4 msg mentioned sender msgTy chName chType
+    return (A.encode notification)
+        where
+            chan = fromJust $ (findChannelById (post^.postChannelIdL) (st^.csChannels))
+            chName = chan^.ccInfo.cdName
+            chType = case (chan^.ccInfo.cdType) of
+                Direct      -> "D"
+                Ordinary    -> "O"
+                Private     -> "P"
+                Group       -> "G"
+                (Unknown t) -> "U"
+            msg = sanitizeUserText $ postMessage post
+            sender = maybePostUsername st post
+            msgTy = case postType post of
```

Comment:
This code appears to duplicate the definition of `msgTy` in the `NotifyV3` case. Would you be willing to factor both (and any other such duplications) out as a top-level function?

Reply:
Sure thing!